### DISCUSSION TITLE
Remove Hyperlink-Decoration on PDF

### DISCUSSION THREAD

**Discussion by speckyspooky** [eclipse-birt/birt#1025] [Q&A] [answered]
Hello There,
I hope some body have an idea of my challenge.
It sounds very easy but I havn't figure out a way to do this.

My task is to add an internal link on an report, export it to PDF and remove the decoration on the hyperlink of PDF.
The decoration replacement of HTML is very easy but I found no way for the Hyperlink on the PDF-document.

Have anybody some hints for me to do the removement on PDF side?

Thanks in advance

**Comment by hvbtup** [comment]
I once was faced with a similar task: The hyperlinks should not be decorated if the PDF is to be printed. Luckily, I know this beforehand, so I decided to just not use hyperlinks in that case.

Like you, I didn't find a way to remove the decoration of a hyperlink. Probably it's hard coded.

**Comment by speckyspooky** [comment]
I figured out a workaround to add an hyperlink without the link style on a text element.
It is not the best case but helpful in my case. The trick is to add the text-element as stadnard element without hyperlink
and afterwards add an "empty text-label" below the standard-text.
The empty-text-element get the hyperlink information and add a negativ margin-top to the empty element.

The effect willbe that the second empty-element give an overlay effect of the text-element above.
So the PDF will show the text and the overlay-empty-text-label will be the clickable PDF-element without link-style.

Currently it is the best workaround to do such stuff.

**Reply by hvbtup** [reply]
So this will work only for those cases where the hyperlink _is_ the full text, not for a dynamic text item where the hyperlink is part of a longer text as in `You should visit the <a href="https://www.eclipse.org">Eclipse web site</a> today`

**Comment by claesrosell** [comment]
Is this to be considered a bug? If so, @speckyspooky, can you file an issue with a small test .rptdesign reproducing this problem? A .rptdesign that has no hyperlink decoration when emitting to HTML, but with the decorations present when emitting a PDF?

**Comment by speckyspooky** [comment, marked as answer]
This topic is added as feature to version BIRT 4.17
